R/validate.R

```diff
@@ -37,6 +38,12 @@ validate_count_mat <- function(count_mat) {
   if (any(is.infinite(count_mat@x))) {
     return(err("matrix values must not be infinite"))
   }
+  if (!all(sapply(barcodes, nzchar))) {
+    return(err("barcodes cannot be the empty string"))
+  }
+  if (!all(sapply(features, nzchar))) {
+    return(err("features cannot be the empty string"))
```

Comment:
other than being strings, is there a more specific format the feature ids should adhere to?

~~also i see the arg is `feature_ids` but we're testing `features` here, is that deliberate?~~
(nvm i see below it's 2 separate things)

Reply:
there's a specific format to ensemble ID we may be able to do some more validation

 https://useast.ensembl.org/Help/Faq?id=488#:~:text=An%20Ensembl%20stable%20ID%20consists,(version).&text=The%20second%20part%20is%20a,(object%20type)(identifier).